### ISSUE TITLE
[request] Bash on Ubuntu on Windows via the Integrated Terminal

### ISSUE THREAD

**Issue by meowklaski** [Microsoft/vscode#10279]
- OS Version: Windows 10 (Anniversary Update)

Now that the Anniversary update is out for Windows 10, it would be great to be able to start a bash session from the integrated terminal. 

I have found that I can run things like ipython without a problem, however...

![capture](https://cloud.githubusercontent.com/assets/11353892/17466247/63d95a54-5cd9-11e6-806a-452ec53d65e7.PNG)

**Comment by Tyriar** [comment]
It can definitely get plugged in using settings https://code.visualstudio.com/docs/editor/integrated-terminal#_configuration, make sure you have installed it and it works in a regular cmd prompt. The support for it is not 100% yet, there are some issues with using the arrow keys for example but you should be able to get it working by adding a setting.

**Comment by crnastena** [comment]
I am not sure that is what the issue is about. I think the ticket is about integrated-terminal not seeing properly installed bash.exe. When intergrated-terminal is open in C:\Windows\System32 folder, and I type dir bash.exe it reports that such file is not found, whereas if I do this from regular command prompt it lists the file. It's like vscode is blocking some exes that are in system32 folder, even from being listed. 

Furthermore, in integrated-terminal dir *.exe list 321 files, whereas regular command prompt list 605.

In summary, we would like to from integrated-terminal that was started as CMD, type in bash and bash shell loads.

**Comment by Tyriar** [comment]
On the folders, first I think `bash.exe` is in `Windows\sysnative` which points to `Windows\syswow64` on 64-bit machines, that may be the reason you're not seeing it in `System32`.

On wanted `bash` to launch the bash on WSL, it's not as easy as just handling `bash` separately. The way the terminal works is that the backing shell received the keystrokes; b, then a, then s, then h, then enter. This is then handled directly by cmd.exe, VS Code has very little to do with this, trying to handle this edge case in VS Code would be very messy and probably not that reliable. This is an upstream issue as far as I can tell, *unless* it's related to the 32-bit version of cmd.exe running by default. I'm curious whether running `bash` with the following integrated terminal setting will work:

```json
"terminal.integrated.shell.windows": "C:\\Windows\\sysnative\\cmd.exe"
```
